File: tests/parsers/samples/code/cs/simple.cs
using System.Collections.Generic;

namespace Samples {
    public static class SimpleMetrics {
        public static IDictionary<string, double> RunSimple() {
            var analyzer = ComplexMetrics.BuildDefault();
            return new Dictionary<string, double> {
                ["baseline"] = analyzer.Distance(10, 3),
                ["rms"] = analyzer.RootMeanSquare(new [] { 1.0, 2.0, 3.0 })
            };
        }

        public static string Summarize() {
            var metrics = RunSimple();
            return $"Summary: baseline={metrics[\"baseline\"]:0.00}, rms={metrics[\"rms\"]:0.00}";
        }
    }
}

File: tests/parsers/samples/code/scala/simple.scala
package samples

case class Metrics(baseline: Double, rms: Double)

object SimpleMetrics {
  def runSimple(): Metrics = {
    val baseline = ComplexMetrics.distance(10, 3, 0.84)
    val rms = ComplexMetrics.rootMeanSquare(List(1.0, 2.0, 3.0))
    Metrics(baseline, rms)
  }

  def summarize(): String = {
    val metrics = runSimple()
    f"baseline=${metrics.baseline}%.2f, rms=${metrics.rms}%.2f"
  }
}
